tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -56,6 +57,15 @@ static DatadogLogging()
                     // Do nothing when an exception is thrown for attempting to access the filesystem
                 }
 
+                try
+                {
+                    Task.Run(() => CleanLogFiles(logDirectory));
+                }
+                catch
+                {
+                    // Do nothing when an exception is thrown for attempting to access the filesystem;
+                }
```

Comment:
You need to move the try catch _inside_ the `CleanLogFiles` method. This `try-catch` only catches exceptions thrown immediately by `Task.Run()`, but you're not observing the Task by calling `await` or `.GetAwaiter().GetResult()` so this could throw an `UnobservedTaskException` I believe too. All fixed by moving the try-catch inside 🙂 

There's an argument to be had as to whether we need to make this cancellable too - probably fine, but might be good to be safe in case of v. slow file system for example.